pupil-spa/src/app/services/storage/storage.service.ts

```diff
@@ -54,4 +54,23 @@ export class StorageService {
       return obj;
     }, {});
   }
+
+  mergeItems(key: StorageKey): any {
+    const localStorageItems = this.getAllItems();
+    const matchingKeys =
+      Object.keys(localStorageItems).filter(lsi => lsi.indexOf(key) >= 0);
+    const sortedMatchingKeys = matchingKeys.sort((a, b) => localStorageItems[a].clientTimestamp - localStorageItems[b].clientTimestamp);
+    const matchingItems = [];
+    sortedMatchingKeys.forEach(smk => {
+      matchingItems.push(localStorageItems[smk]);
+    });
+    localStorage.setItem(key, JSON.stringify(matchingItems));
```

Comment:
This looks like it is creating an array and saving it to `localStorage`.  I am not sure we should run two schemes concurrently.  What is the need?